addons/payment_alipay/controllers/main.py

```diff
@@ -43,18 +43,20 @@ def _alipay_validate_notification(self, **post):
         response = requests.post(alipay.alipay_get_form_action_url(), val)
         response.raise_for_status()
         _logger.info('Validate alipay Notification %s' % response.text)
-        # After program is executed, the page must print “success” (without quote). If not, Alipay server would keep re-sending notification, until over 24 hour 22 minutes Generally, there are 8 notifications within 25 hours (Frequency: 2m,10m,15m,1h,2h,6h,15h)
+        # After program is executed, the page must print “success” (without quote). If not, Alipay server would
+        # keep re-sending notification, until over 24 hour 22 minutes
+        # Generally, there are 8 notifications within 25 hours (Frequency: 2m,10m,15m,1h,2h,6h,15h).
         if response.text == 'true':
             self._alipay_validate_data(**post)
             return 'success'
         return ""
 
     @http.route('/payment/alipay/return', type='http', auth="public", methods=['GET', 'POST'])
-    def alipay_return(self, **post):
+    def return_from_redirect(self, **post):
```

Comment:
```suggestion
    def return_from_redirect(self, **data):
```
Nitpicking: There is a near-0 chance that those data will actually be POST data.